torch/distributed/_state_dict_utils.py

```diff
@@ -142,12 +166,13 @@ def _iterate_state_dict(
         raise ValueError(f"Unexpected value type {type(iter_object)}")
 
     if not ranks_only or dist.get_rank(pg) in ranks_only:
-        if isinstance(ret, torch.Tensor) and cpu_offload:
-            if companion_obj is None:
+        if isinstance(ret, torch.Tensor):
+            if cpu_offload and companion_obj is None:
                 ret = ret.to(cpu_device)
-            else:
+
```

Comment:
So, if `companion_obj` is not None and `cpu_offload` is True, we will automatically use `copy_`. We should comment this logic as `_copy_state_dict` is a standalone API. So `cpu_offload == True and companion_obj is Not None` actually does not exist in our case.

Reply:
Agreed, but quick nit: `cpu_offload` does not change the behavior for `cpu_offload`. If companion_object is provided, it's always assumed the intent is to copy the return value into the companion object. Before this PR, this would only happen if `cpu_offload == True`, but this was misleading because there is no enforcement that the companion object is on the cpu.

Reply:
yup, I just want to ensure people are not confused by the logic.